storagemarket/deal_execution.go

```diff
@@ -50,6 +50,11 @@ func (p *Provider) runDeal(deal *types.ProviderDealState, dh *dealHandler) {
 		p.saveDealToDB(dh.Publisher, deal)
 	}
 
+	// Fail offline deals or retry early if start epoch has passed
+	if err := p.checkDealProposalStartEpoch(deal); err != nil {
+		p.failDeal(dh.Publisher, deal, err, false)
+	}
+
```

Comment:
I don't think we need this. IIUC the call to failDeal inside provider.Start makes this one redundant.

Reply:
This would be useful when offline deal import is called or retry is called manually. They won't go via the start route.